edi/models/edi_backend.py

```diff
@@ -374,8 +399,12 @@ def exchange_receive(self, exchange_record):
         check = self._exchange_receive_check(exchange_record)
         if not check:
             return False
+        state = exchange_record.edi_exchange_state
+        error = False
+        message = None
+        content = None
         try:
-            self._exchange_receive(exchange_record)
+            content = self._exchange_receive(exchange_record)
```

Comment:
move validate data here and raise a specific exception and catch it, same for output generation
then add 1 new state for output like `output_generate_error` and switch to it.